python/semantic_kernel/agents/open_ai/azure_open_ai_assistant_agent.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from collections.abc import AsyncIterable
+from copy import copy
+from typing import TYPE_CHECKING, Any
+
+from openai import AsyncAzureOpenAI
+
+from semantic_kernel.agents.open_ai.open_ai_assistant_base import OpenAIAssistantBase
+from semantic_kernel.agents.open_ai.open_ai_assistant_definition import OpenAIAssistantDefinition
+from semantic_kernel.agents.open_ai.open_ai_assistant_execution_options import OpenAIAssistantExecutionOptions
+from semantic_kernel.agents.open_ai.open_ai_service_configuration import (
+    AzureOpenAIServiceConfiguration,
+)
+from semantic_kernel.const import DEFAULT_SERVICE_NAME
+from semantic_kernel.exceptions.agent_exceptions import AgentInitializationError
+from semantic_kernel.utils.experimental_decorator import experimental_class
+from semantic_kernel.utils.telemetry.user_agent import APP_INFO, prepend_semantic_kernel_to_user_agent
+
+if TYPE_CHECKING:
+    from openai.resources.beta.assistants import Assistant
+
+    from semantic_kernel.kernel import Kernel
+
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class AzureOpenAIAssistantAgent(OpenAIAssistantBase):
```

Comment:
In .NET we aren't differentiating between an _Azure_ and _Non-Azure_ assistant.  Is this separation entirely necessary?

Reply:
We have different service level configuration required for each. It doesn't make sense for us in Python to have one class OpenAI class that picks and chooses from each.